src/ast/json.rs

```diff
@@ -376,23 +413,18 @@ impl<'arena> Ast<'arena> for Json<'arena> {
 
     /* AST EDITING FUNCTIONS */
 
-    fn from_char(&self, c: char) -> Option<Self> {
-        match c {
-            CHAR_TRUE => Some(Json::True),
-            CHAR_FALSE => Some(Json::False),
-            CHAR_NULL => Some(Json::Null),
-            CHAR_ARRAY => Some(Json::Array(vec![])),
-            CHAR_OBJECT => Some(Json::Object(vec![])),
-            CHAR_STRING => Some(Json::Str("".to_string())),
-            _ => None,
+    fn from_class(node_type: Self::Class) -> Self {
+        match node_type {
+            Self::Class::True => Json::True,
+            Self::Class::False => Json::False,
+            Self::Class::Null => Json::Null,
+            Self::Class::Array => Json::Array(vec![]),
+            Self::Class::Object => Json::Object(vec![]),
+            Self::Class::Str => Json::Str("".to_string()),
```

Comment:
Here, `Self::Class` can be replaced with just `Class` (which would refer directly to the `Class` enum).

Reply:
Aha, actually I didn't know we can use `Class` directly.

Reply:
In this example, you can - `Self::Class` refers to the definition, but the actual type `Class` (referring to `json::Class`) is already in scope, so we can use it directly.